src/aggregation/agg_result.rs

```diff
@@ -172,6 +174,10 @@ impl BucketResult {
                 sum_other_doc_count: _,
                 doc_count_error_upper_bound: _,
             } => buckets.iter().map(|bucket| bucket.get_bucket_count()).sum(),
+            BucketResult::Filter(filter_result) => {
+                // add one for the filter bucket itself
+                1 + filter_result.sub_aggregations.get_bucket_count()
```

Comment:
```suggestion
                filter_result.sub_aggregations.get_bucket_count()
```
The bucket count is the user facing bucket count, which filter does not alter

Reply:
Oh! I didn't know that. Thanks. Fixed.